src/gosling-genomic-axis/axis-track.ts

```diff
@@ -549,36 +559,33 @@ function AxisTrack(HGC: any, ...args: any[]): any {
             this.hideOverlaps(this.allTexts);
         }
 
-        hideOverlaps(allTexts: any) {
+        hideOverlaps(allTexts: TickLabelInfo[]) {
             const tree = new RBush<{ minX: number; minY: number; maxX: number; maxY: number }>();
 
-            // store the bounding boxes of the text objects so we can calculate overlaps
-            allTexts.forEach(({ text }: any) => {
-                text.updateTransform();
-                const b = text.getBounds();
-                const m = 5;
-                tree.insert({
-                    minX: b.x - m,
-                    minY: b.y - m,
-                    maxX: b.x + b.width + m * 2,
-                    maxY: b.y + b.height + m * 2
-                });
-            });
-
-            // TODO: possible to reimplement with rbush?
-            (i: number, j: number) => {
-                if (allTexts[i].importance > allTexts[j].importance) {
-                    allTexts[j].text.visible = false;
-                    if (this.options.layout === 'circular' && allTexts[j].rope) {
-                        this.pTicksCircular.removeChild(allTexts[j].rope);
-                    }
-                } else {
-                    allTexts[i].text.visible = false;
-                    if (this.options.layout === 'circular' && allTexts[i].rope) {
-                        this.pTicksCircular.removeChild(allTexts[i].rope);
+            // using bounding boxes of the text objects, calculate overlaps
+            allTexts
```

Comment:
Given that we are calculating overlaps of a small number of labels, I think it's okay to use another library (i.e., rbush). Using rbush, I've implemented a part that calculates overlaps and hide labels. We can consider removing `box-intersect` in the future.